src/components/WalletSelector/authRequestUtils.ts

```diff
@@ -131,17 +136,57 @@ function isRpcSignatureError(error: Record<string, any>) {
   return typeof error === 'object' && error !== null && 'code' in error && error.code === 4001;
 }
 
+// todo this will be diff for mainnet
+const GNOSIS_SIGN_CONTRACT_ADDRESS = '0x60facecd4dbf14f1ae647afc3d1d071b1c29ace4'
 async function signMessage(
   address: string,
   nonce: string,
   signer: JsonRpcSigner,
   connector: AbstractConnector,
+  library?: Web3Provider,
 ): Promise<Signature> {
   try {
     if (connector instanceof WalletConnectConnector) {
+      
       // This keeps the nonce message intact instead of encrypting it for WalletConnect users
       // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-call
-      return await connector.walletConnectProvider.connector.signPersonalMessage([nonce, address]) as Signature;
+      const signature = await connector.walletConnectProvider.connector.signPersonalMessage([nonce, address]) as Signature;
+      // for gnosis, signature will always be 0x. see Notion for explanation
+
+      // todo create separate signMessage func for gnosis/smart contracts.
+
+      if (library){
```

Comment:
random q: do you have prettier turned on?

Reply:
turned it on now, should be resolved!